substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrTypeRepository.java

```diff
@@ -176,7 +176,8 @@ private void visitModule(TypeInfo typeInfo, Module module) {
     }
 
     private void visitClassLoader(TypeInfo typeInfo, ClassLoader classLoader) {
-        if (classLoader != null && typeInfo.addClassLoader(classLoader)) {
+        // Note: null class-loader is ok, we serialize it as "bootstrap" class-loader.
+        if (typeInfo.addClassLoader(classLoader) && classLoader != null) {
```

Comment:
I think there was a misunderstanding, sorry for the lack of clarity. The "boostrap" is when `classLoader.getName()` is null, not when the `classLoader` is null.

Reply:
Okay looks like I was wrong on the second pass with this comment and actually right with the first comment. Sorry about that. So TLDR, ignore this comment chain.